mmv1/products/tpuv2/QueuedResource.yaml

```diff
@@ -0,0 +1,110 @@
+# Copyright 2024 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+name: 'QueuedResource'
+description: |
+  A Cloud TPU Queued Resource.
+min_version: 'beta'
+references:
+  guides:
+    'Official Documentation': 'https://cloud.google.com/tpu/docs/'
+  api: 'https://cloud.google.com/tpu/docs/reference/rest/v2/projects.locations.queuedResources'
+base_url: 'projects/{{project}}/locations/{{zone}}/queuedResources'
+self_link: 'projects/{{project}}/locations/{{zone}}/queuedResources/{{name}}'
+create_url: 'projects/{{project}}/locations/{{zone}}/queuedResources?queuedResourceId={{name}}'
+update_url: 'projects/{{project}}/locations/{{zone}}/queuedResources/{{name}}'
+update_verb: 'PATCH'
+update_mask: true
+autogen_async: true
+async:
+  actions: ['create', 'delete', 'update']
+  type: 'OpAsync'
+  operation:
+    base_url: '{{op_id}}'
+    path: 'name'
+    wait_ms: 1000
+  result:
+    path: 'response'
+    resource_inside_response: true
+  status:
+    path: 'done'
+    complete: true
+    allowed:
+      - true
+      - false
+  error:
+    path: 'error'
+    message: 'message'
+examples:
+  - name: 'tpu_v2_queued_resource_basic'
+    primary_resource_id: 'qr'
+    min_version: 'beta'
+    vars:
+      qr_name: 'test-qr'
+      tpu_name: 'test-tpu'
+    test_env_vars:
+      project: 'PROJECT_NAME'
+    skip_vcr: true
+parameters:
```

Comment:
I think you also need to add `project` to the list of parameters. Currently the resource is able to use the default project value that users can set in the Google provider but they're unable to set a project value through the configuration for this resource.

Reply:
I don't think the resources need the project parameter specifically. Looking at other resources, I don't see any other resources with project as a parameter, yet they are able to specify it in their examples and `project` still shows up on the Terraform resource documentation. Compute Engine Instances are one example of [no project parameter or property](https://github.com/GoogleCloudPlatform/magic-modules/blob/39b904188e53f162d10708cf35da3d48cd55731a/mmv1/products/compute/Instance.yaml#L55) yet [Terraform docs say project can still be configured for this resource](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/compute_instance#project-1). Another example is [firebasedatabase Instance which doesn't have the project parameter](https://github.com/GoogleCloudPlatform/magic-modules/blob/39b904188e53f162d10708cf35da3d48cd55731a/mmv1/products/firebasedatabase/Instance.yaml#L70) yet [project can also be configured for this resource](https://registry.terraform.io/providers/hashicorp/google-beta/latest/docs/resources/firebase_database_instance#project-2) and [is present in one of the examples with the resource](https://github.com/GoogleCloudPlatform/magic-modules/blob/0e4f8103665a42efd12b0c474388d44f6ef112c5/mmv1/templates/terraform/examples/firebase_database_instance_basic.tf.tmpl#L3).

I updated the queued resource example with a project field to show in the example that the project can be configured for this resource.

Reply:
Thanks for looking into this - it looks like answer to the problem is that the code generator adds in a project field automatically if the generator detects that the base_url value contains `{{project}}`, departing from what is described in the YAML file. For example [the link to the diff for the google-beta in this diff report](https://github.com/GoogleCloudPlatform/magic-modules/pull/12277#issuecomment-2465592744) shows that the project field is present in the schema of the new resource you are adding despite it not being in the YAML in this PR. As the generator is stopping the problem I described happening I agree you don't need to explicitly include it in the schema.